pinot-controller/src/main/java/org/apache/pinot/controller/BaseControllerStarter.java

```diff
@@ -493,7 +493,7 @@ protected void configure() {
 
   private ServiceStatus.ServiceStatusCallback generateServiceStatusCallback(HelixManager helixManager) {
     return new ServiceStatus.ServiceStatusCallback() {
-      private boolean _isStarted = false;
+      private volatile boolean _isStarted = false;
```

Comment:
Good catch. `_statusDescription` might also need to be `volatile`

Reply:
Ack